### DISCUSSION TITLE
Java query returns different results in CLI and Visual Code

### DISCUSSION THREAD

**Discussion by yuval-piiano** [github/codeql#12449] [Q&A] [answered]
Hi,

I am running on a single database, a java data flow query. 
When using the visual code extension I am getting 230 results, 
but when running "codeql database analyze" on the same database, qlpacks.yml and query-  from the exact same folder - I am getting only 72 results. 

Anyone has a clue why that happens? How can I get the complete results from the cli? am I missing some step ?

**Comment by aibaars** [comment, marked as answer]
I think `codeql database analyze` groups alerts that are reported on the same location. Have you tried running with `--no-group-results` ?
```
      --no-group-results     [SARIF formats only] Produce one result per
                               message, rather than one result per unique
                               location.
```
